tests/Properties.hs

```diff
@@ -293,6 +301,10 @@ tests = testGroup "properties" [
             , testProperty "ObjectWithSingleField" (toParseJSON gNullaryParseJSONObjectWithSingleField gNullaryToJSONObjectWithSingleField)
             ]
         ]
+      , testGroup "EitherTextInt" [
+        --  testProperty "UntaggedValue" (isUntaggedValue . gEitherTextIntToJSONUntaggedValue)
+        --, testProperty "roundtrip" (toParseJSON gEitherTextIntParseJSONUntaggedValue gEitherTextIntToJSONUntaggedValue)
```

Comment:
Oh, of course. I don't think adding generics support would be too complicated, especially since `UntaggedValue` is conceptually very simple. I don't have time to concoct something right now, but if you want to take a shot at it I can review later.